.github/workflows/check_verilog.yml

```diff
@@ -0,0 +1,95 @@
+name: check-verilog
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+
+env:
+  CIRCT_VERSION: 1.72.0
+  CIRCT_ARCHIVE: "circt-full-shared-linux-x64.tar.gz"
+  DEPENDENCIES_DIR: "deps"
+  MLIR_TARGETS_PATH: "lib/cmake/mlir/MLIRTargets.cmake"
+  BUILD_DIR: "build"
+  UTOPIA_EXECUTABLE: "src/umain"
+  MODULE_OUTPUT_PATH: "output.sv"
+  LIBRARY_OUTPUT_PATH: "lib.sv"
+
+jobs:
+  build-and-parse-verilog:
```

Comment:
First of all, not "verilog" but "systemverilog", or, probably, "sv".

Also this job name looks a bit straightforward. What if one more step, (for example, logic synthesis) will be added here? What name will be used -- "build-and-parse-and-synthesize-systemverilog"? I'd use more general name like "synth-sv" -- our main task is to perform good HLS, and we can omit extra details here and write a describing comment, if needed,